### PR TITLE
Add missing documentation entries to IChallengeNonce interfaces

### PR DESCRIPTION
**Pull request by erkkiarus**
Signed-off-by: Erkki Arus <erkki@raulwalter.com>

### CODE REVIEW

File: src/WebEid.Security/WebEid.Security.csproj [hunk 1]
```diff
@@ -25,6 +26,14 @@
     <LangVersion>8.0</LangVersion>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
+    <NoWarn>1701;1702;VSSpell001;VSSpell002</NoWarn>
```
Comment (rabadashTheFool) [inline src/WebEid.Security/WebEid.Security.csproj:30]: Why do we need to disable warning for these codes? Maybe you can provide an example. On Windows with Visual Studio I have no issues without this.

Comment (erkkiarus) [reply src/WebEid.Security/WebEid.Security.csproj:30]: About 1701 and 1702 I'm not sure, but VSSpell-warnings I can comment. For long time, when you wanted to use a spell-checker, you had to download separated extension for Visual Studio. At some point, MS integrated a spell-checker into Visual Studio 2022. Unlike the extension, this one checked the spelling also in type, identifier and variable names. Class and variable names rarely are correct English words and these were marked as misspelled words. That start to be annoying and therefore we added VSSpell001 and VSSpell002 into warnings exclusion list.

Comment (rabadashTheFool) [reply src/WebEid.Security/WebEid.Security.csproj:30]: Fair point regarding the naming conventions. Based on your feedback, I conducted some research and testing.

I experimented with the native Spell Checker in Visual Studio 2022. When enabled, it does not recognize 'auth' as a valid word, among other instances. However, adding the `<NoWarn>` element to my .csproj file does not influence this behavior in Visual Studio, as the VSSpell001 and VSSpell002 warnings pertain solely to the Speller extension.
Consequently, with Visual Studio 2022's native Spell Checker, the `<NoWarn>` entries are rendered superfluous.

Could you try disabling the extension and removing these entries from the .csproj file to see if it resolves the issue?

Regarding CS1701 and CS1702:
- **CS1701**: This warning may indicate a version mismatch between two assembly references, potentially requiring a runtime policy in the application's .config file to rectify.
- **CS1702**: Similar to CS1701, this warning points to a possible discrepancy in build and/or revision numbers between assembly references, which might not automatically unify.

To my knowledge, our Solution does not exhibit these issues, rendering these warnings unnecessary.

Comment (erkkiarus) [reply src/WebEid.Security/WebEid.Security.csproj:30]: No, it did not go better. You can enable the Visual Studio's built-in Spell Checker from _Edit->Advanced->Toggle Spell Checking_ menu item and you will see, it starts marking identifiers as misspelled words.
However, I'll remove the warnings from projects `<NoWarn>` property and let every developed decide himself what to do.

File: src/WebEid.Security/WebEid.Security.csproj [hunk 2]
```diff
@@ -25,6 +26,14 @@
...
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
+    <NoWarn>1701;1702;VSSpell001;VSSpell002</NoWarn>
+  </PropertyGroup>
+
```
Comment (rabadashTheFool) [inline src/WebEid.Security/WebEid.Security.csproj:36]:
```suggestion

```

File: src/WebEid.Security/WebEid.Security.csproj (hunk 1 above)
Comment (rabadashTheFool) [reply src/WebEid.Security/WebEid.Security.csproj]: Removed with suggestion.

File: src/WebEid.Security/WebEid.Security.csproj [hunk 3]
```diff
@@ -12,6 +12,7 @@
     <Authors>RIA</Authors>
     <Copyright>Copyright © Republic of Estonia Information System Authority 2023</Copyright>
     <Version>1.1.0</Version>
+    <FileVersion>1.1.2.1</FileVersion>
```
Comment (rabadashTheFool) [inline src/WebEid.Security/WebEid.Security.csproj:15]:
```suggestion
```

**Review by mrts — APPROVED** [review]

**Pull request merged**